network.c

```diff
@@ -1402,6 +1402,29 @@ struct iio_context * network_create_context(const char *host)
 #endif
 	{
 		ret = getaddrinfo(host, IIOD_PORT_STR, &hints, &res);
+		/*
+		 * It might be an avahi hostname which means that getaddrinfo() will only work if
+		 * nss-mdns is installed on the host and /etc/nsswitch.conf is correctly configured
+		 * which might be not the case for some minimalist distros. In this case,
+		 * as a last resort, let's try to resolve the host with avahi...
+		 */
+		if (ret) {
+			char addr_str[DNS_SD_ADDRESS_STR_MAX];
+
+			IIO_DEBUG("'getaddrinfo()' failed: %s. Trying dnssd as a last resort...\n",
+				  gai_strerror(ret));
+			ret = dnssd_resolve_host(host, addr_str, sizeof(addr_str));
+			if (ret) {
+				char buf[512];
```

Comment:
512 bytes seems a lot for an error message; 128 or 256?

Reply:
I'm already reducing it... it's 1024 some lines above this. But yeah, most likely we don't really need 512 bytes